src/tracing/mod.rs

```diff
@@ -355,6 +348,25 @@ impl TracingInspector {
             })
             .expect("is valid opcode;");
 
+        let memory = self
+            .config
+            .record_memory_snapshots
+            .then(|| modifies_memory(op))
+            .and_then(|is_memory_modified| {
```

Comment:
we need to record memory snapshots per step.
so what need to do here is:

1. change `RecordedMemory(Vec<u8>)` to `RecordedMemory(Bytes)`
2. if the opcode does not modify memory, get the recorded memory snapshot from the previous recorded step, if it exist, see referenced foundry code